app/Exceptions/Handler.php

```diff
@@ -49,18 +51,43 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
-        if (! $request->expectsJson()) {
-            return parent::render($request, $exception);
+        if (strpos($request->getUri(), '/api/v') !== false) {
+            if ($exception instanceof NotFoundHttpException) {
+                return response()->json([
+                    'message' => 'Requested route is invalid.',
+                ], 400);
+            } elseif ($exception instanceof AuthenticationException) {
+                return response()->json([
+                    'message' => 'Unable to authenticate request, check if your access token is correct.',
+                ], 401);
+            } elseif ($exception instanceof HttpException) {
+                if ($exception->getMessage() == 'Too Many Attempts.') {
+                    return response()->json([
+                        'message' => "The rate limit has been exceeded, please wait before sending more requests",
+                    ], 429);
+                } else {
+                    //Maybe this should be logged?
```

Comment:
I thought a bit more about this and I believe every exception is being logged at the moment, maybe I should make it so only unexpected errors get logged?

Reply:
Actually we can even improve the logs by connecting to a 3rd party service like Sentry, which I think is a very reasonable idea for the near future.

Besides the info from the exception itself, in some cases we might need to additional something more, we'll have to go case by case.

Logs should be handled on the `report` method.

https://laravel.com/docs/5.5/errors#report-method

Reply:
I totally agree that logging should be improved soon, maybe another issue should be created to address that? I'm not sure how complex integration with a 3rd party is but it might keep things more organized

Reply:
The service is already prepared to be integrated with Laravel apps, it is quite straightforward.

I have opened an issue for this matter #10